modules/consensus/accept.go

```diff
@@ -135,16 +114,27 @@ func (s *State) acceptBlock(b types.Block) error {
 	// happens before checking that the transactions are intrinsically valid
 	// because it's a much cheaper operation for us to verify, and it's
 	// expensive for an attacker to spoof the header.
-	err := s.validHeader(b)
+	err := cs.validHeader(b)
 	if err != nil {
 		return err
 	}
 
 	// Try adding the block to the tree.
-	err = s.addBlockToTree(b)
+	revertedNodes, appliedNodes, err := cs.addBlockToTree(b)
 	if err != nil {
 		return err
 	}
+	if len(appliedNodes) > 0 {
+		cs.updateSubscribers(revertedNodes, appliedNodes)
+	}
+
+	// Sanity check, if applied nodes is len 0, revertedNodes should also be
+	// len 0.
+	if build.DEBUG {
+		if len(appliedNodes) == 0 && len(revertedNodes) != 0 {
+			panic("appliedNodes and revertedNodes are mismatched!")
+		}
+	}
```

Comment:
that could clean things up a bit.

assertion := if ... && ... || ...
build.Assert(assertion, ErrAssertionFailed)